ccmlib/scylla_node.py

```diff
@@ -536,51 +537,78 @@ def start(self, join_ring=True, no_wait=False, verbose=False,
         jvm_args = jvm_args + ['--collectd-hostname',
                                '%s.%s' % (socket.gethostname(), self.name)]
 
-        # Let's add jvm_args and the translated args
-
-        args = [launch_bin, '--options-file', options_file, '--log-to-stdout', '1'] + jvm_args + translated_args
-
-        # Lets search for default overrides in SCYLLA_EXT_OPTS
-        scylla_ext_opts = os.getenv('SCYLLA_EXT_OPTS', "").split()
-        opts_i = 0
-        orig_args = list(args)
-        while opts_i < len(scylla_ext_opts):
-            if scylla_ext_opts[opts_i].startswith("--scylla-manager="):
-               opts_i += 1
-            elif scylla_ext_opts[opts_i].startswith('-'):
-                o = scylla_ext_opts[opts_i]
+        args = [launch_bin, '--options-file', options_file, '--log-to-stdout', '1']
+
+        MB = 1024 * 1024
+
+        def process_opts(opts, set_during_test = False):
+            # holds all extended args except --smp and --memory
+            # that are handled separately
+            ext_args = OrderedDict()
+            # the --memory option is just used at this stage
+            # to derive self._mem_mb_per_cpu from memory / smp
+            memory_opt = None
+            opts_i = 0
+            while opts_i < len(opts):
+                # the command line options show up either like "--foo value-of-foo"
+                # or as a single option like --yes-i-insist
+                assert opts[opts_i].startswith('-')
+                if opts[opts_i].startswith("--scylla-manager="):
+                    opts_i += 1
+                    continue
+                o = opts[opts_i]
                 opts_i += 1
                 if '=' in o:
                     opt = o.replace('=', ' ', 1).split()
+                    key = opt[0]
+                    val = opt[1]
                 else:
-                    opt = [ o ]
-                    while opts_i < len(scylla_ext_opts) and not scylla_ext_opts[opts_i].startswith('-'):
-                        opt.append(scylla_ext_opts[opts_i])
+                    key = o
+                    vals = []
+                    while opts_i < len(opts) and not opts[opts_i].startswith('-'):
+                        vals.append(opts[opts_i])
                         opts_i += 1
-                if opt[0] not in orig_args:
-                    args.extend(opt)
+                    val = ' '.join(vals)
+                if key not in ext_args:
+                    if key == '--smp':
+                        self._smp = int(val)
+                        if set_during_test:
+                            # sticky, only set to True
+                            self._smp_set_during_test = True
```

Comment:
i think the reason why we have this flag is to provide the setting with highest priority if a test sets the option using methods like `set_smp()` or `set_mem_mb_per_cpu()`. but this change set this flag when we set these options in command line option. does this defeat the purpose of having these flags?

Reply:
It was supposed to be `set_during_test=False` when processing the environment.
However, with this series, there's no use for these flags anymore so I'll just remove them in the next version.